File: 435-non-overlapping-intervals/non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        removed = 0
        last_valid = None

        intervals = sorted(intervals, key=lambda interval:interval[1])

        for idx, interval in enumerate(intervals):
            if not last_valid or last_valid[1] <= interval[0]:
                last_valid = interval
            else:
                removed += 1
        
        return removed
            

File: 435-non-overlapping-intervals/non-overlapping-intervals.go
func eraseOverlapIntervals(intervals [][]int) int {
    sort.Slice(intervals, func(i, j int) bool {
        return intervals[i][1] < intervals[j][1]
    })

    var maxEnd int
    count := 0

    for i, interval := range(intervals) {
        if i == 0 || maxEnd <= interval[0] {
            count += 1
            maxEnd = interval[1]
        }
    }

    return len(intervals) - count
}